### PR TITLE
Feature/basic pub sub example

### PR DESCRIPTION
**Pull request by rosemwahlin**
1) Added XML QoS files
2) Fixed build errors/warnings (found while setting up initial continuous integration for Linux/Windows builds)

### CODE REVIEW

File: 1_hello_world/c++11/application.hpp [hunk 1]
```diff
@@ -34,7 +35,7 @@ inline void setup_signal_handlers()
     signal(SIGTERM, stop_handler);
 }
 
-enum class ParseReturn { OK, ERROR, EXIT };
+enum class ParseReturn { OK, FAILURE, EXIT };
```
Comment (fgaranda) [inline 1_hello_world/c++11/application.hpp]: Nice! I forgot to mention that in the code review! Pretty the use of protected enums.

File: 1_hello_world/c++11/USER_QOS_PROFILES.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    file.
+
+    The profile in this file inherits from the builtin QoS profile
+    BuiltinQosLib::Generic.StrictReliable. That profile, along with all of the
+    other built-in QoS profiles can be found in the
+    BuiltinProfiles.documentationONLY.xml file located in the
+    $NDDSHOME/resource/xml/ directory.
+-->
+<dds xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+     xsi:noNamespaceSchemaLocation="/Applications/rti_connext_dds-6.0.1/resource/schema/rti_dds_qos_profiles.xsd">
```
Comment (fgaranda) [inline 1_hello_world/c++11/USER_QOS_PROFILES.xml]: You should point the community portal. It's probably best to use the "current" path.

https://community.rti.com/schema/current/rti_dds_qos_profiles.xsd

We used to have latest, I don't know which term is best.

File: 1_hello_world/c++11/USER_QOS_PROFILES.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0"?>
+
```
Comment (fgaranda) [inline 1_hello_world/c++11/USER_QOS_PROFILES.xml]: Would it look nicer without that space and an extra space at the end, with two spaces or with none? Not sure.

File: 1_hello_world/c++98/USER_QOS_PROFILES.xml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    The profile in this file inherits from the builtin QoS profile
+    BuiltinQosLib::Generic.StrictReliable. That profile, along with all of the
+    other built-in QoS profiles can be found in the
+    BuiltinProfiles.documentationONLY.xml file located in the
+    $NDDSHOME/resource/xml/ directory.
+-->
+<dds xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+     xsi:noNamespaceSchemaLocation="/Applications/rti_connext_dds-6.0.1/resource/schema/rti_dds_qos_profiles.xsd">
+    <!--
```
Comment (fgaranda) [inline 1_hello_world/c++98/USER_QOS_PROFILES.xml]: Same comment as above.

File: 1_hello_world/c++98/application.h [hunk 5]
```diff
@@ -33,7 +33,7 @@ inline void setup_signal_handlers()
     signal(SIGTERM, stop_handler);
 }
 
-enum ParseReturn { OK, ERROR, EXIT };
+enum ParseReturn { OK, FAILURE, EXIT };
```
Comment (fgaranda) [inline 1_hello_world/c++98/application.h]: I think OK and FAILURE are still too common. We could protect them here with something like `PARSE_RETURN_OK`, `PARSE_RETURN_FAILURE`, and `PARSE_RETURN_EXIT`.

Comment (rosemwahlin) [reply 1_hello_world/c++98/application.h]: Done.

File: 1_hello_world/c++98/USER_QOS_PROFILES.xml (hunk 4 above)
Comment (rosemwahlin) [reply 1_hello_world/c++98/USER_QOS_PROFILES.xml]: Done.

File: 1_hello_world/c++11/USER_QOS_PROFILES.xml (hunk 3 above)
Comment (rosemwahlin) [reply 1_hello_world/c++11/USER_QOS_PROFILES.xml]: I'm not really sure, either. I'm going to leave as-is for now.

File: 1_hello_world/c++11/USER_QOS_PROFILES.xml (hunk 2 above)
Comment (rosemwahlin) [reply 1_hello_world/c++11/USER_QOS_PROFILES.xml]: Done.

File: 1_hello_world/c++11/application.hpp (hunk 1 above)
Comment (rosemwahlin) [reply 1_hello_world/c++11/application.hpp]: Changed it to the more-specific text you recommended :)

**Pull request merged**